src/classes/class-opcache-stats.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Niteo\WooCart\Defaults\OpCacheStats {
+
+	use \WooCart\Log\Socket;
+
+	/**
+	 * Class Reporter
+	 *
+	 * @package Niteo\WooCart\Defaults\OpCacheStats
+	 */
+	class Reporter {
+
+		const cron_hook = 'WOOCart\OpCacheStats\Reporter';
+
+
+		public function __construct() {
+
+			if ( ! wp_next_scheduled( self::cron_hook ) ) {
+				wp_schedule_event( time(), 'hourly', self::cron_hook );
+			}
+
+			add_action( self::cron_hook, [ $this, 'report_stats' ] );
+		}
+
+		/**
+		 * Hook function that runs as part of cron.
+		 *
+		 * @return bool
+		 *
+		 * @codeCoverageIgnore
+		 */
+		public function report_stats(): bool {
+			$plugins = $this->parse_plugins( opcache_get_status() );
+			$plugins = $this->decorate( $plugins );
+			return $this->emit( $plugins );
+		}
+
+		/**
+		 * Parse opcache stats and sum stats by plugin.
+		 *
+		 * @param array $report
+		 * @return array
+		 */
+		public function parse_plugins( array $report ): array {
+			$scripts        = $report['scripts'];
+			$wp_plugin_path = wp_normalize_path( WP_PLUGIN_DIR );
+
+			// plugins path + /
+			$plugins_base_length = strlen( $wp_plugin_path ) + 1;
+			$plugins             = [];
+			foreach ( $scripts as $script => $stats ) {
+
+				// script should be a plugin
+				if ( ! strstr( $script, $wp_plugin_path ) ) {
+					continue;
+				}
+				// left trim so it becomes `plugin-slug/{index.php, ...}`
+				$plugin_base = substr( $script, $plugins_base_length, $plugins_base_length );
```

Comment:
```suggestion
				$plugin_base = substr( "wp-content/plugins/super-long-plugin-name-123456789/...", 19, 19 );
				// "super-long-plugin-na"
```
I'm wondering if this can happen because you use `$plugins_base_length` as substring length

Reply:
Ah no, because both cache and opcache have absolute paths. If one, for example, uses chroot, then both paths would be again based on the same folder structure.